[BUG] Pylint CI job needs to install all dependencies first

# Bug Description

```
************* Module ols.user_data_collection.data_collector
ols/user_data_collection/data_collector.py:22:0: E0401: Unable to import 'kubernetes' (import-error)
ols/user_data_collection/data_collector.py:23:0: E0401: Unable to import 'requests' (import-error)
************* Module ols.app.main
ols/app/main.py:6:0: E0401: Unable to import 'fastapi' (import-error)
ols/app/main.py:7:0: E0401: Unable to import 'starlette.datastructures' (import-error)
ols/app/main.py:8:0: E0401: Unable to import 'starlette.responses' (import-error)
************* Module ols.app.routers
ols/app/routers.py:3:0: E0401: Unable to import 'fastapi' (import-error)
************* Module ols.app.models.config
ols/app/models/config.py:8:0: E0401: Unable to import 'pydantic' (import-error)
************* Module ols.app.models.models
ols/app/models/models.py:6:0: E0401: Unable to import 'pydantic' (import-error)
ols/app/models/models.py:7:0: E0401: Unable to import 'pydantic.dataclasses' (import-error)
************* Module ols.app.metrics.token_counter
ols/app/metrics/token_counter.py:6:0: E0401: Unable to import 'langchain.callbacks.base' (import-error)
ols/app/metrics/token_counter.py:7:0: E0401: Unable to import 'langchain.llms.base' (import-error)
ols/app/metrics/token_counter.py:8:0: E0401: Unable to import 'langchain_core.outputs.llm_result' (import-error)
************* Module ols.app.metrics.metrics
ols/app/metrics/metrics.py:5:0: E0401: Unable to import 'fastapi' (import-error)
ols/app/metrics/metrics.py:6:0: E0401: Unable to import 'fastapi.responses' (import-error)
ols/app/metrics/metrics.py:7:0: E0401: Unable to import 'prometheus_client' (import-error)
************* Module ols.app.endpoints.health
ols/app/endpoints/health.py:12:0: E0401: Unable to import 'fastapi' (import-error)
ols/app/endpoints/health.py:13:0: E0401: Unable to import 'langchain_core.messages.ai' (import-error)
************* Module ols.app.endpoints.ols
ols/app/endpoints/ols.py:12:0: E0401: Unable to import 'pytz' (import-error)
ols/app/endpoints/ols.py:13:0: E0401: Unable to import 'fastapi' (import-error)
************* Module ols.app.endpoints.feedback
ols/app/endpoints/feedback.py:9:0: E0401: Unable to import 'fastapi' (import-error)
************* Module ols.app.endpoints.authorized
ols/app/endpoints/authorized.py:7:0: E0401: Unable to import 'fastapi' (import-error)
************* Module ols.src.prompts.prompt_generator
ols/src/prompts/prompt_generator.py:3:0: E0401: Unable to import 'langchain_core.messages' (import-error)
ols/src/prompts/prompt_generator.py:4:0: E0401: Unable to import 'langchain_core.prompts' (import-error)
************* Module ols.src.auth.k8s
ols/src/auth/k8s.py:7:0: E0401: Unable to import 'kubernetes.client' (import-error)
ols/src/auth/k8s.py:8:0: E0401: Unable to import 'fastapi' (import-error)
ols/src/auth/k8s.py:9:0: E0401: Unable to import 'kubernetes.client.rest' (import-error)
ols/src/auth/k8s.py:10:0: E0401: Unable to import 'kubernetes.config' (import-error)
************* Module ols.src.auth.auth_dependency_interface
ols/src/auth/auth_dependency_interface.py:5:0: E0401: Unable to import 'fastapi' (import-error)
************* Module ols.src.auth.noop
ols/src/auth/noop.py:5:0: E0401: Unable to import 'fastapi' (import-error)
************* Module ols.src.rag_index.index_loader
ols/src/rag_index/index_loader.py:27:4: E0401: Unable to import 'llama_index.core' (import-error)
ols/src/rag_index/index_loader.py:28:4: E0401: Unable to import 'llama_index.core.embeddings.utils' (import-error)
ols/src/rag_index/index_loader.py:29:4: E0401: Unable to import 'llama_index.core.indices.base' (import-error)
ols/src/rag_index/index_loader.py:30:4: E0401: Unable to import 'llama_index.core.llms.utils' (import-error)
ols/src/rag_index/index_loader.py:31:4: E0401: Unable to import 'llama_index.vector_stores.faiss' (import-error)
ols/src/rag_index/index_loader.py:59:12: E0401: Unable to import 'llama_index.embeddings.huggingface' (import-error)
************* Module ols.src.query_helpers.docs_summarizer
ols/src/query_helpers/docs_summarizer.py:6:0: E0401: Unable to import 'langchain.chains' (import-error)
ols/src/query_helpers/docs_summarizer.py:7:0: E0401: Unable to import 'llama_index.core' (import-error)
************* Module ols.src.query_helpers.query_docs
ols/src/query_helpers/query_docs.py:6:0: E0401: Unable to import 'langchain_core.documents.base' (import-error)
ols/src/query_helpers/query_docs.py:7:0: E0401: Unable to import 'langchain_core.vectorstores' (import-error)
************* Module ols.src.query_helpers.question_validator
ols/src/query_helpers/question_validator.py:6:0: E0401: Unable to import 'langchain.chains' (import-error)
ols/src/query_helpers/question_validator.py:7:0: E0401: Unable to import 'langchain.prompts' (import-error)
************* Module ols.src.query_helpers.query_helper
ols/src/query_helpers/query_helper.py:7:0: E0401: Unable to import 'langchain.llms.base' (import-error)
************* Module ols.src.query_helpers.attachment_appender
ols/src/query_helpers/attachment_appender.py:5:0: E0401: Unable to import 'yaml' (import-error)
************* Module ols.src.ui.gradio_ui
ols/src/ui/gradio_ui.py:7:0: E0401: Unable to import 'gradio' (import-error)
ols/src/ui/gradio_ui.py:8:0: E0401: Unable to import 'requests' (import-error)
ols/src/ui/gradio_ui.py:9:0: E0401: Unable to import 'fastapi' (import-error)
************* Module ols.src.cache.redis_cache
ols/src/cache/redis_cache.py:7:0: E0401: Unable to import 'redis' (import-error)
ols/src/cache/redis_cache.py:8:0: E0401: Unable to import 'redis.backoff' (import-error)
ols/src/cache/redis_cache.py:9:0: E0401: Unable to import 'redis.exceptions' (import-error)
ols/src/cache/redis_cache.py:14:0: E0401: Unable to import 'redis.retry' (import-error)
************* Module ols.src.cache.postgres_cache
ols/src/cache/postgres_cache.py:7:0: E0401: Unable to import 'psycopg2' (import-error)
************* Module ols.src.llms.llm_loader
ols/src/llms/llm_loader.py:6:0: E0401: Unable to import 'langchain.llms.base' (import-error)
************* Module ols.src.llms.providers.fake_provider
ols/src/llms/providers/fake_provider.py:6:0: E0401: Unable to import 'langchain.llms.base' (import-error)
ols/src/llms/providers/fake_provider.py:7:0: E0401: Unable to import 'langchain_community.llms' (import-error)
************* Module ols.src.llms.providers.azure_openai
ols/src/llms/providers/azure_openai.py:7:0: E0401: Unable to import 'azure.core.credentials' (import-error)
ols/src/llms/providers/azure_openai.py:8:0: E0401: Unable to import 'azure.identity' (import-error)
ols/src/llms/providers/azure_openai.py:9:0: E0401: Unable to import 'langchain.llms.base' (import-error)
ols/src/llms/providers/azure_openai.py:10:0: E0401: Unable to import 'langchain_openai' (import-error)
************* Module ols.src.llms.providers.bam
ols/src/llms/providers/bam.py:6:0: E0401: Unable to import 'genai' (import-error)
ols/src/llms/providers/bam.py:7:0: E0401: Unable to import 'genai.extensions.langchain' (import-error)
ols/src/llms/providers/bam.py:8:0: E0401: Unable to import 'genai.schema' (import-error)
ols/src/llms/providers/bam.py:9:0: E0401: Unable to import 'langchain.llms.base' (import-error)
************* Module ols.src.llms.providers.provider
ols/src/llms/providers/provider.py:9:0: E0401: Unable to import 'httpx' (import-error)
ols/src/llms/providers/provider.py:10:0: E0401: Unable to import 'ibm_watsonx_ai.metanames' (import-error)
ols/src/llms/providers/provider.py:13:0: E0401: Unable to import 'langchain.llms.base' (import-error)
************* Module ols.src.llms.providers.rhelai_vllm
ols/src/llms/providers/rhelai_vllm.py:6:0: E0401: Unable to import 'langchain.llms.base' (import-error)
ols/src/llms/providers/rhelai_vllm.py:7:0: E0401: Unable to import 'langchain_openai' (import-error)
************* Module ols.src.llms.providers.openai
ols/src/llms/providers/openai.py:6:0: E0401: Unable to import 'langchain.llms.base' (import-error)
ols/src/llms/providers/openai.py:7:0: E0401: Unable to import 'langchain_openai' (import-error)
************* Module ols.src.llms.providers.watsonx
ols/src/llms/providers/watsonx.py:6:0: E0401: Unable to import 'ibm_watsonx_ai.metanames' (import-error)
ols/src/llms/providers/watsonx.py:9:0: E0401: Unable to import 'langchain.llms.base' (import-error)
ols/src/llms/providers/watsonx.py:10:0: E0401: Unable to import 'langchain_ibm.llms' (import-error)
************* Module ols.src.llms.providers.rhoai_vllm
ols/src/llms/providers/rhoai_vllm.py:6:0: E0401: Unable to import 'langchain.llms.base' (import-error)
ols/src/llms/providers/rhoai_vllm.py:7:0: E0401: Unable to import 'langchain_openai' (import-error)
************* Module ols.customize.ols.reranker
ols/customize/ols/reranker.py:5:0: E0401: Unable to import 'llama_index.core.schema' (import-error)
************* Module ols.utils.certificates
ols/utils/certificates.py:8:0: E0401: Unable to import 'certifi' (import-error)
ols/utils/certificates.py:9:0: E0401: Unable to import 'cryptography' (import-error)
************* Module ols.utils.pyroscope
ols/utils/pyroscope.py:7:0: E0401: Unable to import 'requests' (import-error)
ols/utils/pyroscope.py:25:12: E0401: Unable to import 'pyroscope' (import-error)
************* Module ols.utils.errors_parsing
ols/utils/errors_parsing.py:5:0: E0401: Unable to import 'fastapi' (import-error)
ols/utils/errors_parsing.py:6:0: E0401: Unable to import 'genai.exceptions' (import-error)
ols/utils/errors_parsing.py:7:0: E0401: Unable to import 'ibm_watsonx_ai.wml_client_error' (import-error)
ols/utils/errors_parsing.py:8:0: E0401: Unable to import 'openai' (import-error)
************* Module ols.utils.config
ols/utils/config.py:7:0: E0401: Unable to import 'yaml' (import-error)
************* Module ols.utils.checks
ols/utils/checks.py:8:0: E0401: Unable to import 'pydantic' (import-error)
************* Module ols.utils.token_handler
ols/utils/token_handler.py:6:0: E0401: Unable to import 'llama_index.core.schema' (import-error)
ols/utils/token_handler.py:7:0: E0401: Unable to import 'tiktoken' (import-error)
************* Module ols.runners.uvicorn
ols/runners/uvicorn.py:5:0: E0401: Unable to import 'uvicorn' (import-error)
************* Module scripts.generate_openapi_schema
scripts/generate_openapi_schema.py:7:0: E0401: Unable to import 'fastapi.openapi.utils' (import-error)

-----------------------------------
Your code has been rated at 8.46/10
```

--- a/.github/workflows/pylint.yaml
+++ b/.github/workflows/pylint.yaml
@@ -14,7 +14,11 @@ jobs:
           python-version: '3.11'
       - name: Python version
         run: python --version
-      - name: Python linter installation
-        run: pip install --user pylint
+      - name: PDM installation
+        run: pip install --user pdm
+      - name: Install dependencies
+        run: pdm install
+      - name: Install devel dependencies
+        run: pdm install --dev
       - name: Python linter
-        run: pylint ols/ scripts/
+        run: pdm run pylint ols/ scripts/ tests/ runner.py

--- a/scripts/evaluation/response_evaluation.py
+++ b/scripts/evaluation/response_evaluation.py
@@ -1,0 +1,16 @@
+"""Response evaluation using pre-defined question/answer pair."""
+
+
+class ResponseEvaluation:
+    """Evaluate LLM response."""
+
+    def __init__(self, eval_args, api_client):
+        """Initialize."""
+
+    def validate_response(self):
+        """Validate LLM response."""
+        return True
+
+    def evaluate_models(self):
+        """Evaluate models against groundtruth answer."""
+        return None

--- a/scripts/upload_artifact_s3.py
+++ b/scripts/upload_artifact_s3.py
@@ -1,0 +1,17 @@
+"""Upload artifact containing the pytest results and configuration to an s3 bucket.
+
+This will in turn get picked by narberachka service,
+which will either send it to ibutsu or report portal, based on the
+prefix of the file name.
+A dictionary containing the credentials of the S3 bucket must be specified, containing the keys:
+AWS_BUCKET
+AWS_REGION
+AWS_ACCESS_KEY_ID
+AWS_SECRET_ACCESS_KEY
+
+"""
+
+
+def upload_artifact_s3(aws_env):
+    """Upload artifact to the specified S3 bucket."""
+    return True

